cli/coverage_cmd.go

```diff
@@ -0,0 +1,186 @@
+package cli
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	"sourcegraph.com/sourcegraph/go-flags"
+
+	"sourcegraph.com/sourcegraph/srclib/config"
+	"sourcegraph.com/sourcegraph/srclib/graph"
+	"sourcegraph.com/sourcegraph/srclib/grapher"
+	"sourcegraph.com/sourcegraph/srclib/plan"
+)
+
+func init() {
+	cliInit = append(cliInit, func(cli *flags.Command) {
+		_, err := cli.AddCommand("coverage",
+			"srclib coverage",
+			"compute approximate amount of code successfully analyzed by srclib",
+			&coverageCmd,
+		)
+		if err != nil {
+			log.Fatal(err)
+		}
+	})
+}
+
+type Coverage struct {
+	FileScore      float32  // % files successfully processed
+	RefScore       float32  // % internal refs that resolve to a def
+	TokDensity     float32  // average number of refs/defs per LoC
+	UncoveredFiles []string // files for which srclib data was not successfully generated (best-effort guess)
+}
+
+type codeFileDatum struct {
+	LoC          int
+	NumRefs      int
+	NumDefs      int
+	NumRefsValid int
+}
+
+type CoverageCmd struct {
+}
+
+var coverageCmd CoverageCmd
+
+func (c *CoverageCmd) Execute(args []string) error {
+	repo, err := OpenLocalRepo()
+	if err != nil {
+		return err
+	}
+
+	cvg, err := coverage(repo)
+	if err != nil {
+		return err
+	}
+
+	out, err := json.MarshalIndent(cvg, "", "  ")
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(out))
+
+	return nil
+}
+
+var codeExts = []string{".go", ".java", ".py", ".rb", ".cpp", ".ts", ".cs"} // codeExt lists all file extensions that indicate a code file we want to cover
```

Comment:
Yes.



    _____________________________
From: Jon Poler <notifications@github.com>
Sent: Sunday, February 28, 2016 6:32 PM
Subject: Re: [srclib] Coverage cmd (#254)
To: sourcegraph/srclib <srclib@noreply.github.com>
Cc: Beyang Liu <beyang@sourcegraph.com>


    

In cli/coverage_cmd.go:   > +> +	cvg, err := coverage(repo)> +	if err != nil {> +		return err> +	}> +> +	out, err := json.MarshalIndent(cvg, "", "  ")> +	if err != nil {> +		return err> +	}> +	fmt.Println(string(out))> +> +	return nil> +}> +> +var codeExts = []string{".go", ".java", ".py", ".rb", ".cpp", ".ts", ".cs"} // codeExt lists all file extensions that indicate a code file we want to cover   

Is .java the only extension for java code? (I'm asking out of Ignorance.)   

—
Reply to this email directly or view it on GitHub.